### ISSUE TITLE
[QUESTION] What do we think about null-safety?

### ISSUE THREAD

**Issue by FrankTN** [CollActionteam/collaction_app#5]
**Problem**
Null safety is not implemented in a few dependencies yet (listed below), so I have to run the app in PyCharm with the `--no-sound-null-safety` flag. I would like to discuss what we can do about this, personally, the idea of null safety sounds like a nice feature to have so I think we should work towards enforcing it, but a significant amount of libraries don't seem to be supporting it yet. Haven't tried any of the migration tools listed in the link below yet. Would love to hear what everyone thinks.

**Error Message:**
Error: Cannot run with sound null safety, because the following dependencies
don't support null safety:

 - package:graphql_flutter
 - package:graphql
 - package:connectivity
 - package:hive
 - package:path_provider
 - package:normalize
 - package:gql_exec
 - package:gql_link
 - package:gql
 - package:http
 - package:connectivity_platform_interface
 - package:crypto
 - package:path_provider_linux
 - package:path_provider_windows
 - package:path_provider_platform_interface
 - package:gql_http_link
 - package:gql_error_link
 - package:gql_dedupe_link
 - package:gql_transform_link
 - package:plugin_platform_interface
 - package:xdg_directories
 - package:websocket
 - package:rxdart
 - package:uuid_enhanced
 - package:http_parser
 - package:convert

For solutions, see https://dart.dev/go/unsound-null-safety

**Comment by rubenhorn** [comment]
I agree that we should try to have sound null safety and we will probably introduce more non null safe code in the future.   Regarding how to deal with it I propose the following strategy:

- If possible, all code must be null safe (enforced by reviewer)
- If not (trivially) possible, this must be addressed in a new / added to an existing tech debt task
- Tech debt tasks must rise in priority over time (fix old stuff before adding new stuff)

TLDR; New Jira task for next sprint? ✌😅

**Comment by FrankTN** [comment]
For now updating the packages to their latest versions has worked, so I think we can mark this issue as resolved. Perhaps we can reopen if we encounter this issue again at a later stage.
